Dans cette partie, il nous est demandé de déterminer les 3 premiers pics du diffractogramme correspondant à notre cristal et de précisier les indices hkl associés. Pour toute cette partie, on considère un rayonnement X incident Cu Kα de longueur d’onde λ=1.54060 Angström. Ces rayons X vont créer des phénomènes d'interférences constructives ou destructives selon la direction. Sur base du modèle de diffraction, nous pouvons touver la position de chacun des atomes du cristal (en respectant certaines hypothèses).

Dans la suite de l'exercie, nous considèrerons $2\theta$ comme l'angle entre la direction d'incidence et celle d'observation (voir image ci-dessous).

<img src="https://github.com/sderaedemack/lmapr1492/blob/master/Images/2Theta.png?raw=true" width="300">

La cellule utilisée pour cet exercice est la cellule primitive.

In [3]:
from pymatgen.ext.matproj import MPRester

with MPRester ("z5aIUAkpFPTHndsr") as m:
    
    #Création de la structure du cristal à partir du MP
    structure = m.get_structure_by_material_id("mp-15783")

In [4]:
from pymatgen.analysis.diffraction.xrd import XRDCalculator

#Création d'un objet XRDCalculator
xrd = XRDCalculator(1.54060)

#Calcul du modèle de diffraction
pat = xrd.get_pattern(structure)

Dans un premier temps, nous pouvons observer le diffractogramme de notre cristal. Celui-ci représente l'angle $2\theta$ en degré en fonction de l'intensité mise à l'échelle (de 0 à 100).

In [5]:
#Affichage du diffractogramme 
xrd.show_plot(structure,annotate_peaks=False,two_theta_range=[0,175])

<Figure size 1600x1000 with 1 Axes>

# Premier pic <h1>

In [9]:
#Récupération de l'angle du premier pic
theta1 = pat.x[0]

#Récupération de l'intensité du premier pic
I1 = pat.y[0]

#Récupération des indices hkl du premier pic
hkl1 = pat.hkls[0]

print('Angle du premier pic : ' + str(theta1) + ' degrés')
print('Intensité du premier pic : ' + str(I1))
print('Indices hkl du premier pic :' + str(hkl1))

Angle du premier pic : 11.973340554041021 degrés
Intensité du premier pic : 68.24713081677788
Indices hkl du premier pic :[{'hkl': (1, 1, 1), 'multiplicity': 2}]


On peut voir que le premier pic se trouve à un angle $2\theta = 11.97334055$ degrés et a une intensité de $68.24713081677788$. Les indices hkl correspondant sont $(1,1,1)$ et ils possèdent une multiplicité de 2.

# Deuxième pic <h1>

In [10]:
#Récupération de l'angle du deuxième pic
theta2 = pat.x[1]

#Récupération de l'intensité du deuxième pic
I2 = pat.y[1]

#Récupération des indices hkl du deuxième pic
hkl2 = pat.hkls[1]

print('Angle du deuxième pic : ' + str(theta2) + ' degrés')
print('Intensité du deuxième pic : ' + str(I2))
print('Indices hkl du deuxième pic :' + str(hkl2))

Angle du deuxième pic : 24.079960017812258 degrés
Intensité du deuxième pic : 15.201938121575763
Indices hkl du deuxième pic :[{'hkl': (2, 2, 2), 'multiplicity': 2}]


On peut voir que le deuxième pic se trouve à un angle $2\theta = 24.079960017812258$ degrés et a une intensité de $15.201938121575763$. Les indices hkl correspondant sont $(2, 2, 2)$ et ils possèdent une multiplicité de 2.

# Troisième pic <h1>

In [11]:
#Récupération de l'angle du troisième pic
theta3 = pat.x[2]

#Récupération de l'intensité du troisième pic
I3 = pat.y[2]

#Récupération des indices hkl du troisième pic
hkl3 = pat.hkls[2]

print('Angle du troisième pic : ' + str(theta3) + ' degrés')
print('Intensité du troisième pic : ' + str(I3))
print('Indices hkl du troisième pic :' + str(hkl3))

Angle du troisième pic : 25.104591995005084 degrés
Intensité du troisième pic : 23.86766555216653
Indices hkl du troisième pic :[{'hkl': (1, 0, 0), 'multiplicity': 6}]


On peut voir que le troisème pic se trouve à un angle $2\theta = 25.104591995005084$ degrés et a une intensité de $23.86766555216653$. Les indices hkl correspondant sont $(1, 0, 0)$ et ils possèdent une multiplicité de 6.

# Références <h1>

**Image:**

LMPAR1492 : Physique des matériaux (Gian-Marco Rignanese)